libraries/standard/http/include/http_client.h

```diff
@@ -0,0 +1,339 @@
+#ifndef HTTP_CLIENT_H_
+#define HTTP_CLIENT_H_
+
+#include <stdint.h>
+#include <stdbool.h>
+
+/**
+ * @brief Maximum amount of headers allowed from the server.
+ * 
+ * If the server sends more headers than this configuration, then status code
+ * #HTTP_SECURITY_ALERT_TOO_MANY_HEADERS is returned from #HTTPClient_Send.
+ */
+#ifndef HTTP_MAX_HEADERS_BYTES
+    #define HTTP_MAX_HEADERS_BYTES      2048U
+#endif
+
+/**
+ * @brief The HTTP header "User-Agent" value.
+ * 
+ * The following headerline is automatically written to
+ * #HTTPRequestHeaders_t.pBuffer:
+ * "User-Agent: AWS\r\n"
+ */ 
+#ifndef HTTP_USER_AGENT_VALUE
+    #define HTTP_USER_AGENT_VALUE       "AWS"
+#endif
+
+/**
+ * Supported HTTP request methods.
+ */
+#define HTTP_METHOD_GET                 "GET"  /**< HTTP Method GET. */
+#define HTTP_METHOD_PUT                 "PUT"  /**< HTTP Method GET. */
+#define HTTP_METHOD_POST                "POST" /**< HTTP Method GET. */
+#define HTTP_METHOD_HEAD                "HEAD" /**< HTTP Method GET. */
+
+
+/**
+ * Flags for #HTTPRequestInfo_t.flags.
+ * These headers control what headers are written or not to the 
+ * #HttpRequestHeaders_t.pBuffer.
+ */
+/**
+ * @brief Set this flag to indicate the request is for a persistent connection.
+ * 
+ * Setting this will cause a "Connection: Keep-Alive" to be written to the
+ * request.
+ * 
+ * Keep in mind some third-party servers do not have to honor persistent
+ * connection requests.
+ */
+#define HTTP_REQUEST_KEEP_ALIVE_FLAG                    0x1U
+/** 
+ * @brief Set this flag to disable automatically writing the Content-Length
+ * header. 
+ */
+#define HTTP_REQUEST_DISABLE_AUTO_CONTENT_LENGTH_FLAG   0x2U
```

Comment:
Why not remove the `AUTO` word from the macro name as `HTTP_REQUEST_DISABLE_CONTENT_LENGTH_HEADER`? What is the significance of the _auto_ word in the functionality? If it just disables the population of the Content-Length header in the `httpClientInitializeHeaders` function, I don't think the word auto is required.

Reply:
Sure I am fine with that.